lib/cucumber/term/banner.rb

```diff
@@ -0,0 +1,58 @@
+require 'cucumber/term/ansicolor'
+
+module Cucumber
+  module Term
+    module Banner
+      def display_banner(lines, io, border_modifiers = nil)
+        BannerMaker.new.display_banner(lines, io, border_modifiers || [:cyan])
+      end
+
+      class BannerMaker
+        include Term::ANSIColor
+
+        def display_banner(lines, io, border_modifiers)
+          lines = lines.split("\n") if lines.is_a? String
+          longest_line_length = lines.map { |line| line_length(line) }.max
+
+          io.puts apply_modifiers("┌#{'─' * (longest_line_length + 2)}┐", border_modifiers)
+          lines.map do |line|
+            padding = ' ' * (longest_line_length - line_length(line))
+            io.puts "#{apply_modifiers('│', border_modifiers)} #{display_line(line)}#{padding} #{apply_modifiers('│', border_modifiers)}"
+          end
+          io.puts apply_modifiers("└#{'─' * (longest_line_length + 2)}┘", border_modifiers)
+        end
+
+        private
+
+        def display_line(line)
+          line.is_a?(Array) ? line.map { |span| display_span(span) }.join : line
+        end
+
+        def display_span(span)
+          return apply_modifiers(span.shift, span) if span.is_a?(Array)
+          span
+        end
+
+        def apply_modifiers(str, modifiers)
+          display = str
+          modifiers.each { |modifier| display = send(modifier, display) }
+          display
+        end
+
+        def line_length(line)
+          if line.is_a?(Array)
```

Comment:
if you flip this conditional you can lose some lines.

```
return line.length unless line.is_a?(Array)

# other logic without if/end
```

Reply:
I prefer staying away from unless when possible because I find it harder to read. I also like having `else` when both conditions branch are equivalent (when there are no guard clauses).

So I rewrote it like this:

```ruby
if line.is_a?(Array)
  line.map { |span| span_length(span) }.sum
else
  line.length
end
```

Reply:
That's fine, but the code is a little inconsistent, so I tried to pick the one you went for most. I'm happy with either way I guess, but you've already merged it.

Reply:
Yes, it is a little inconsistent. Each developer code style evolves over time. Rubocop is here to help us with it thanksfully.